lib/export/dwca/data.rb

```diff
@@ -629,37 +887,664 @@ def biological_associations_resource_relationship_tmp
+          LEFT JOIN temp_media_image_links links ON links.image_id = img.id
+
+          -- Join attribution data
+          LEFT JOIN attributions ON attributions.attribution_object_id = img.id
+                                 AND attributions.attribution_object_type = 'Image'
+
+          -- Join identifiers for UUID and URI
+          LEFT JOIN identifiers uuid_id ON uuid_id.identifier_object_id = img.id
+                                        AND uuid_id.identifier_object_type = 'Image'
+                                        AND uuid_id.type = 'Identifier::Global::Uuid'
+          LEFT JOIN identifiers uri_id ON uri_id.identifier_object_id = img.id
+                                       AND uri_id.identifier_object_type = 'Image'
+                                       AND uri_id.type = 'Identifier::Global::Uri'
+
+          -- Aggregate attribution data using LATERAL joins
+          #{attribution_owners_lateral_sql}
+          #{attribution_creators_lateral_sql}
+          #{attribution_creator_ids_lateral_sql}
+          #{attribution_copyright_holders_lateral_sql}
+
+          -- Filter to only images in the collected set (using temp table to avoid huge IN clause)
+          INNER JOIN temp_media_image_links img_filter ON img_filter.image_id = img.id
+
+          WHERE dwc.\"occurrenceID\" IS NOT NULL  -- Only include media with valid occurrence links
+          ORDER BY img.id
+        ) TO STDOUT WITH (FORMAT CSV, DELIMITER E'\\t', NULL '')
+      SQL
+
+      conn.raw_connection.copy_data(image_copy_sql) do
+        while row = conn.raw_connection.get_copy_data
+          output_file.write(row.force_encoding(Encoding::UTF_8))
+        end
+      end
+    end
+
+    # Exports sound media records to the output file using PostgreSQL COPY TO
+    # Streams data directly to avoid loading entire dataset into memory
+    def export_sounds_to_file(sound_ids, output_file)
+      return if sound_ids.empty?
+
+      conn = ActiveRecord::Base.connection
+      license_case = creative_commons_license_case_sql
+      copyright_label = copyright_label_sql
+
+      sound_copy_sql = <<-SQL
+        COPY (
+          SELECT
+            REGEXP_REPLACE(dwc.\"occurrenceID\", E'[\\n\\t]', ' ', 'g') AS coreid,
+            REGEXP_REPLACE(
+              CASE
+                WHEN uuid_id.identifier IS NOT NULL THEN 'sound:' || uuid_id.identifier
+                WHEN uri_id.identifier IS NOT NULL THEN 'sound:' || uri_id.identifier
+                ELSE 'sound:' || snd.id::text
+              END,
+              E'[\\n\\t]', ' ', 'g'
+            ) AS identifier,
+            'Sound' AS \"dc:type\",
+            snd.id AS \"providerManagedID\",
+            #{license_case} AS \"dc:rights\",
+            #{license_case} AS \"dcterms:rights\",
+            REGEXP_REPLACE(owners.names, E'[\\n\\t]', ' ', 'g') AS \"Owner\",
+            NULL AS \"UsageTerms\",
+            REGEXP_REPLACE(#{copyright_label}, E'[\\n\\t]', ' ', 'g') AS \"Credit\",
```

Comment:
If most of these aren't needed (REGEXP_REPLACE) then the collective string becomes more managable.